agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -298,15 +297,14 @@ func (cs *CredentialSpecResource) GetName() string {
 // Create is used to create all the credentialspec resources for a given task
 func (cs *CredentialSpecResource) Create() error {
 	var err error
-	// To fail fast, check execution role first
+	var iamCredentials credentials.IAMRoleCredentials
+
 	executionCredentials, ok := cs.credentialsManager.GetTaskCredentials(cs.getExecutionCredentialsID())
-	if !ok {
-		// No need to log here. managedTask.applyResourceState already does that
-		err = errors.New("credentialspec resource: unable to find execution role credentials")
-		cs.setTerminalReason(err.Error())
-		return err
+	if ok {
+		iamCredentials = executionCredentials.GetIAMRoleCredentials()
+	} else {
+		seelog.Debugf("Unable to obtain execution credentials: %v", ok)
```

Comment:
We do not want to return an error here as the credentials are only required for S3/SSM and not the file option.

Reply:
Then I think you should skip this check when it is for file on host and keep this check before you require the S3/SSM. We want customer to get clear error message when task fails instead that they need to turn on debug log for agent in order to know what is happening.

Reply:
Fail fast is definitely preferred over a debug log which customers would be struggling to figure out what went wrong

Reply:
We cannot fail fast in this scenario. The credentials are required only for S3/SSM option. 
Also we do not want to obtain the credentials again and again for the same task across multiple containers. CP already enforces the credentials when S3/SSM option is used.

Reply:
I think you can remove the debug log here and check whether iamCredentials are nil or not in **handleS3CredentialspecFile/handleSSMCredentialspecFile** and return error inside it. It will not cause confusion for customers when they are using credentialSpec on the host, besides, it can also error out when it is execution role error.  When agent restarts, the execution credentials may not exist, we want to explicitly catch this situation and don't treat them same as other runtime error.

Reply:
what happens for some reason role is not present for s3/ssm option. 
We cannot always assume this will be the case only for local file.

Reply:
CP has validation to ensure that for S3/SSM we will always have valid credentials. 
However, if we somehow end up with invalid creds, it should be a runtime failure. 

I have updated the PR with new changes.

Reply:
there are cases where agent could restart in the middle of resource creation. in that case, agent might end up not having the required credentials. So this check will be needed. I suggest you test this use case as well.

Reply:
please follow @yumex93 's suggestion for fixing this.

Reply:
nvm, just saw its fixed